linkerd/app/inbound/src/direct.rs

```diff
@@ -120,6 +128,16 @@ impl<N> Inbound<N> {
                 .push(transport::metrics::NewServer::layer(
                     rt.metrics.proxy.transport.clone(),
                 ))
+                .check_new_service::<AuthorizedLocalTcp, _>()
+                .push_map_target(|(permit, tcp): (policy::ServerPermit, LocalTcp)| {
+                    AuthorizedLocalTcp {
+                        addr: tcp.server_addr,
+                        client_id: tcp.client_id,
+                        permit,
+                    }
+                })
+                .check_new_service::<(policy::ServerPermit, LocalTcp), _>()
+                .push(policy::NewTcpPolicy::layer(rt.metrics.tcp_authz.clone()))
```

Comment:
IMO it feels kind of like it would be nicer to me if the state transition from `LocalTcp` -> `AuthorizedLocalTcp` happened _inside_ of the `NewTcpPolicy` layer...but, i guess trying to abstract over types that can be turned into an authorized target type by adding a `ServerPermit` with some kind of trait is probably not worth the complexity, so, this is fine IMO...